packages/react-dom/src/events/__tests__/SyntheticEvent-test.js

```diff
@@ -9,114 +9,224 @@
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+    container = null;
   });
 
   it('should normalize `target` from the nativeEvent', () => {
-    var target = document.createElement('div');
-    var syntheticEvent = createEvent({srcElement: target});
+    var instance;
+    var expectedCount = 0;
+
+    var eventHandler = syntheticEvent => {
+      expect(syntheticEvent.target).toBe(instance);
+
+      expectedCount++;
+    };
+    instance = ReactDOM.render(<div onClick={eventHandler} />, container);
+
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    // Emulate IE8
+    Object.defineProperty(event, 'target', {
+      get() {},
+    });
+    Object.defineProperty(event, 'srcElement', {
+      get() {
+        return instance;
+      },
+    });
+    instance.dispatchEvent(event);
 
-    expect(syntheticEvent.target).toBe(target);
-    expect(syntheticEvent.type).toBe(undefined);
+    expect(expectedCount).toBe(1);
   });
 
   it('should be able to `preventDefault`', () => {
-    var nativeEvent = {};
-    var syntheticEvent = createEvent(nativeEvent);
+    var instance;
+    var expectedCount = 0;
 
-    expect(syntheticEvent.isDefaultPrevented()).toBe(false);
-    syntheticEvent.preventDefault();
-    expect(syntheticEvent.isDefaultPrevented()).toBe(true);
+    var eventHandler = syntheticEvent => {
+      expect(syntheticEvent.isDefaultPrevented()).toBe(false);
+      syntheticEvent.preventDefault();
+      expect(syntheticEvent.isDefaultPrevented()).toBe(true);
+      expect(syntheticEvent.defaultPrevented).toBe(true);
 
-    expect(syntheticEvent.defaultPrevented).toBe(true);
+      expectedCount++;
+    };
+    instance = ReactDOM.render(<div onClick={eventHandler} />, container);
 
-    expect(nativeEvent.returnValue).toBe(false);
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    instance.dispatchEvent(event);
+
+    expect(expectedCount).toBe(1);
   });
 
   it('should be prevented if nativeEvent is prevented', () => {
-    expect(createEvent({defaultPrevented: true}).isDefaultPrevented()).toBe(
-      true,
-    );
-    expect(createEvent({returnValue: false}).isDefaultPrevented()).toBe(true);
+    var instance;
+    var expectedCount = 0;
+
+    var eventHandler = syntheticEvent => {
+      expect(syntheticEvent.isDefaultPrevented()).toBe(true);
+
+      expectedCount++;
+    };
+    instance = ReactDOM.render(<div onClick={eventHandler} />, container);
+
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    Object.defineProperty(event, 'defaultPrevented', {
+      get() {
+        return true;
+      },
+    });
+    instance.dispatchEvent(event);
+
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    // Emulate IE8
```

Comment:
What does this test have to do with IE8?

The test title says "should be prevented if nativeEvent is prevented". To test this, remove these properties and just call `e.nativeEvent.preventDefault()`. Then check `e.isDefaultPrevented()`.

Reply:
The previous version of this test had two expectations. On where `defaultPrevented` was false and one where `returnValue` was false. I checked why and it turned out IE uses `returnValue` instead of `defaultPrevented`. Happy to use `e.nativeEvent.preventDefault()` but it will have less parity with the previous version.